### DISCUSSION TITLE
no-conditional-in-test using isMobile

### DISCUSSION THREAD

**Discussion by Morgy93** [playwright-community/eslint-plugin-playwright#178] [Q&A]
Hello,

regarding the rule [`no-conditional-in-test`](https://github.com/playwright-community/eslint-plugin-playwright/blob/main/docs/rules/no-conditional-in-test.md) I'd like to know if there is some recommended way of using `isMobile` in conditionals:

```
test("click on navigation link", async ({ page, isMobile }) => {

    if (isMobile) {
        ... open the mobile navigation menu    
    }

   ... click on the navigation link called home
});
```

Any suggestions?

**Comment by mskelton** [comment]
My recommendation would be to move that logic into a [page object](https://playwright.dev/docs/pom) so then you can simply call a method such as `openNavLink` which internally does the `if (isMobile) {` check.

**Reply by Morgy93** [reply]
That just moves the issue from one file to another, doesn't it? 🤔

**Reply by mskelton** [reply]
Not really. The `no-conditional-in-test` rule detects if the statement is inside a test block, and warns then. If it doesn't detect you are in a test block (e.g. a page object) it won't warn.
